lib/elastomer/client/delete_by_query.rb

```diff
@@ -0,0 +1,82 @@
+module Elastomer
+  class Client
+
+    class DeleteByQuery
+
+      def initialize( client, query, params = nil )
+        @client    = client
+        @query     = query
+        @params    = params
+        @responses = []
+      end
+
+      attr_reader :client, :query, :params, :responses
+
+      def is_ok( status )
+        status.between?(200, 299)
+      end
+
+      def categorize( items )
+        {
+          "found" => items.count { |i| i["found"] },
+          "deleted" => items.count { |i| is_ok(i["status"]) },
+          "missing" => items.count { |i| !i["found"] },
+          "failed" => items.count { |i| i["found"] && !is_ok(i["status"]) },
+        }
+      end
+
+      def accumulate( response )
+        @responses << response unless response == nil
```

Comment:
In chat we already talked about the response accumulation here. I'm looking forward to the improvements to the bulk operations that will make this process unneeded.

Here is an arcane bit of ruby trivia: `response.nil?` is faster to process than `response == nil`. I prefer the interrogatory methods over equality checks for this reason. And it is more difficult to do the dreaded single `=` mistake `response = nil` in an if statement.